src/Features/Blockcore.Features.WebHost/WebHostFeature.cs

```diff
@@ -0,0 +1,225 @@
+
+            if (this.settings.KeepaliveTimer == null)
+            {
+                this.logger.LogTrace("(-)[KEEPALIVE_DISABLED]");
+                return Task.CompletedTask;
+            }
+
+            // Start the keepalive timer, if set.
+            // If the timer expires, the node will shut down.
+            this.settings.KeepaliveTimer.Elapsed += (sender, args) =>
+            {
+                this.logger.LogInformation($"The application will shut down because the keepalive timer has elapsed.");
+
+                this.settings.KeepaliveTimer.Stop();
+                this.settings.KeepaliveTimer.Enabled = false;
+                this.fullNode.NodeLifetime.StopApplication();
+            };
+
+            this.settings.KeepaliveTimer.Start();
+
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Prints command-line help.
+        /// </summary>
+        /// <param name="network">The network to extract values from.</param>
+        public static void PrintHelp(Network network)
+        {
+            WebHostSettings.PrintHelp(network);
+        }
+
+        /// <summary>
+        /// Get the default configuration.
+        /// </summary>
+        /// <param name="builder">The string builder to add the settings to.</param>
+        /// <param name="network">The network to base the defaults off.</param>
+        public static void BuildDefaultConfigurationFile(StringBuilder builder, Network network)
+        {
+            WebHostSettings.BuildDefaultConfigurationFile(builder, network);
+        }
+
+        /// <inheritdoc />
+        public override void Dispose()
+        {
+            // Make sure the timer is stopped and disposed.
+            if (this.settings.KeepaliveTimer != null)
+            {
+                this.settings.KeepaliveTimer.Stop();
+                this.settings.KeepaliveTimer.Enabled = false;
+                this.settings.KeepaliveTimer.Dispose();
+            }
+
+            // Make sure we are releasing the listening ip address / port.
+            if (this.webHost != null)
+            {
+                this.logger.LogInformation("API stopping on URL '{0}'.", this.settings.ApiUri);
+                this.webHost.StopAsync(TimeSpan.FromSeconds(WebHostStopTimeoutSeconds)).Wait();
+                this.webHost = null;
+            }
+        }
+    }
+
+    public sealed class WebHostFeatureOptions
+    {
+        public IClientEvent[] EventsToHandle { get; set; }
+
+        public (Type Broadcaster, ClientEventBroadcasterSettings clientEventBroadcasterSettings)[] ClientEventBroadcasters { get; set; }
+    }
+
+    /// <summary>
+    /// A class providing extension methods for <see cref="IFullNodeBuilder"/>.
+    /// </summary>
+    public static class ApiFeatureExtension
+    {
+        [Obsolete("Use the new UseWebHost()")]
+        public static IFullNodeBuilder UseApi(this IFullNodeBuilder fullNodeBuilder, Action<WebHostFeatureOptions> optionsAction = null)
```

Comment:
If we are already ditching backwards compatibility?

Reply:
Ditched! Removed!